pkg/registry/scheduling/priorityclass/strategy.go

```diff
@@ -46,16 +45,10 @@ func (priorityClassStrategy) NamespaceScoped() bool {
 func (priorityClassStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	pc := obj.(*scheduling.PriorityClass)
 	pc.Generation = 1
-	schedulingutil.DropDisabledFields(pc, nil)
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
-func (priorityClassStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-	newClass := obj.(*scheduling.PriorityClass)
-	oldClass := old.(*scheduling.PriorityClass)
-
-	schedulingutil.DropDisabledFields(newClass, oldClass)
```

Comment:
Just FYI for @ravisantoshgudimetla: once you start to work on mutable PriorityClass, need to drop in logic to update the generation by 1 here.